pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -115,6 +140,17 @@ func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 func (c *CSIMaxVolumeLimitChecker) filterAttachableVolumes(
 	nodeInfo *schedulernodeinfo.NodeInfo, volumes []v1.Volume, namespace string, result map[string]string) error {
 
+	node := nodeInfo.Node()
+	if node == nil {
+		return nil
+	}
+
+	csiNode, err := c.csiNodeInfo.GetCSINodeInfo(node.Name)
+	if err != nil {
+		klog.V(5).Infof("Could not get a CSINode object for the node: %v", err)
```

Comment:
This is possible for older nodes?

Reply:
Yes, nodes where, for some reason, there's no `CSINode`. On second thought, I think we want to return the error and prevent scheduling the pod (recording the error in the events).